### Урок 5. Домашняя работа

In [1]:
import numpy as np
from math import sqrt

1. Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.

In [2]:
X = 80
sigma = 16
n = 256
Z_a2 = 1.96

In [3]:
X1 = X + Z_a2 * sigma/sqrt(n)
X2 = X - Z_a2 * sigma/sqrt(n)

In [4]:
print([X2, X1])

[78.04, 81.96]


2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

In [5]:
dim = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
dim

array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [6]:
sigma = np.std(dim, ddof=1)
sigma

0.4508017549014448

In [7]:
X = np.mean(dim)
X

6.590000000000001

In [8]:
n = 10
t_a2 = 2.262

In [9]:
X1 = X + t_a2 * sigma/sqrt(n)
X2 = X - t_a2 * sigma/sqrt(n)

In [10]:
print([X2, X1])

[6.267538255912426, 6.912461744087575]


3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

In [11]:
M0 = 17
M = 17.5
sigma = 2
n = 100
alpha = 0.05
Z = 1.96 # Находим по таблице накопленного нормального распредделения

* *H0: M = M0*
* *H1: M > M0*

In [12]:
Zн = (M - M0) / (sigma / sqrt(n))
Zн

2.5

Так как 2.5 > 1.65 мы попадаем в область H1, соответственно верна альтернативная гипотеза

4. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [13]:
weight = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
weight

array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [14]:
M = weight.mean()
M

198.5

* *H0: M = M0*
* *H1: M != M0*

Так как сигма неизвестна тестировать будем по Стьюденту

In [15]:
M0 = 200
n = 10
alpha = 0.005
T_t1 = - 3.25 # по алфа находим значение в таблице
T_t2 = 3.25

In [16]:
sigma_n = np.std(weight, ddof=1)
sigma_n

4.453463071962462

In [17]:
T = (M - M0) / (sigma_n / sqrt(n))
T

-1.0651074037450896

Так как -1.065 > - 3.25 значение находится в области H0, соответственно верна H0 гипотеза